### PR TITLE
Clear molecule molgraph cache on reset

### PR DESCRIPTION
**Pull request by KnathanM**
`MoleculeDataset` has a `reset` method that un-does any potential scaling to the extra atom/bond features/descriptors. It is used in `__post_init__` to populate the public versions. We don't really use it elsewhere. 

But the other day, I was testing if a model could overfit to some data. I had a single dataset that had extra atom features. I had scaled them and given the scaler to the model. During training, chemprop expects scaled extra atom features, but during prediction, it expects them to be not scaled and will automatically scale them. Rather than make my dataset again, I tried calling `reset` on it to undo the scaling to use it as a test set. This didn't work though because I had also cached the molgraphs and the cached molgraphs had the scaled extra atom features. 

I therefore propose that `reset` also refreshes the molgraph cache, similar to what happens when you try to set the V_f and E_f directly. 

We don't need to call `_init_cache` in `MolAtomBondDataset` because it calls `super().reset()` and the other properties it resets are not saved in the molgraphs. `ReactionDataset` doesn't take V_f and E_f, so we don't need to worry about this for it.

### CODE REVIEW

**Comment by KnathanM** [comment]
My original idea had a problem because reset was called before cache was set, so reset couldn't call _init_cache. I tried setting cache before the reset, but then V_f and E_f aren't populated so the cache on the fly can't be initialized. 

I'm going to close this. I think it is reasonable to say a user needs to manually reset the cache after calling reset if that is what they want to do:
```
dataset.reset()
dataset.cache = True
```
